app/src/main/java/com/asfoundation/wallet/ui/SettingsFragment.kt

```diff
@@ -142,7 +148,7 @@ class SettingsFragment : PreferenceFragmentCompat(), SettingsView {
     val walletPreference = findPreference<Preference>("pref_wallet")
     walletPreference?.summary = walletAddress
     walletPreference?.setOnPreferenceClickListener {
-      manageWalletsRouter.open(activity, false)
+      context?.let { manageWalletsRouter.open(it) }
```

Comment:
**TO DISCUSS:**

Shouldn't this validation be done inside manageWalletsRouter? 
This would remove extra validations and logic from the view

Reply:
agree